rtl/eclrtl/rtlnewkey.cpp

```diff
@@ -0,0 +1,2133 @@
+/*##############################################################################
+
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR getValue OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include <initializer_list>
+#include "jlib.hpp"
+#include "jsort.hpp"
+#include "jexcept.hpp"
+#include "rtlnewkey.hpp"
+#include "eclrtl_imp.hpp"
+#include "rtlrecord.hpp"
+#include "rtlkey.hpp"
+
+//move to jstring
+static void readString(StringBuffer &out, const char * &in)
+{
+    for (;;)
+    {
+        char c = *in++;
+        if (!c)
+            throw MakeStringException(0, "Invalid filter - missing closing '");
+        if (c=='\'')
+            break;
+        if (c=='\\')
+        {
+            c = *in++;
+            switch (c)
+            {
+            case '\'':
+            case '\\':
+                break;
+            default:
+                UNIMPLEMENTED;
```

Comment:
Is there a Jira open to complete this work? Would it be easier just to do it now? Is there any existing code that we can common up to do it?